src/utils/staking.js

```diff
@@ -275,23 +277,39 @@ export class Staking {
     }
 
     async getValidators(accountIds) {
+        const { current_validators, next_validators, current_proposals } = await this.provider.validators()
+        const currentValidators = current_validators.map(({ account_id }) => account_id)
+        const nextValidators = current_proposals.map(({ account_id }) => account_id)
+        
         if (!accountIds) {
-            const { current_validators, next_validators, current_proposals } = await this.provider.validators()
-            const allValidators = [...current_validators, ...next_validators, ...current_proposals]
-            accountIds = [...new Set(allValidators.map(({ account_id }) => account_id))]
+            const rpcValidators = [...current_validators, ...next_validators, ...current_proposals].map(({ account_id }) => account_id)
+
+            // TODO use indexer - getting all historic validators from raw GH script .json
+            const networkId = this.provider.connection.url.indexOf('mainnet') > -1 ? 'mainnet' : 'testnet'
+            if (!ghValidators) {
+                ghValidators = (await fetch(`https://raw.githubusercontent.com/frol/near-validators-scoreboard/scoreboard-${networkId}/validators_scoreboard.json`).then((r) => r.json()))
+                .map(({ account_id }) => account_id)
+            }
+            
+            accountIds = [...new Set([...rpcValidators, ...ghValidators])]
+                .filter((v) => v.indexOf('nfvalidator') === -1 && v.indexOf(networkId === 'mainnet' ? '.near' : '.m0') > -1)
         }
+
         return (await Promise.all(
             accountIds.map(async (account_id) => {
-                const validator = {
-                    accountId: account_id,
-                    contract: await this.getContractInstance(account_id, stakingMethods)
-                }
                 try {
+                    const validator = {
+                        accountId: account_id,
+                        current: currentValidators.includes(account_id),
+                        next: nextValidators.includes(account_id),
+                        contract: await this.getContractInstance(account_id, stakingMethods)
+                    }
                     const fee = validator.fee = await validator.contract.get_reward_fee_fraction()
                     fee.percentage = fee.numerator / fee.denominator * 100
                     return validator
                 } catch (e) {
-                    if (!/cannot find contract code|wasm execution failed/.test(e.message)) {
+                    console.log('error with', account_id)
```

Comment:
makes sense to log error itself? Also probably should use `console.warn`. Should we log in case of expected errors however? Seem like only unexpected would be useful.